src/lib/operators/jit_operator/jit_aware_lqp_translator.cpp

```diff
@@ -49,94 +90,92 @@ std::shared_ptr<JitOperatorWrapper> JitAwareLQPTranslator::_try_translate_node_t
     // aggregate nodes that would be placed in the middle of an operator chain.
     const auto aggregate_node = std::static_pointer_cast<AggregateNode>(node);
 
     auto aggregate = std::make_shared<JitAggregate>();
-    const auto column_names = aggregate_node->output_column_names();
-    const auto groupby_columns = aggregate_node->groupby_column_references();
-    const auto aggregate_columns = aggregate_node->aggregate_expressions();
-
-    // Split the output column names between groupby columns and aggregate expressions.
-    // The AggregateNode::_update_output() function node will always output groupby columns first.
-    const auto groupby_column_names = boost::adaptors::slice(column_names, 0, groupby_columns.size());
-    const auto aggregate_column_names =
-        boost::adaptors::slice(column_names, groupby_columns.size(), column_names.size());
-
-    for (const auto& groupby_column : boost::combine(groupby_column_names, groupby_columns)) {
-      const auto expression = _try_translate_column_to_jit_expression(groupby_column.get<1>(), *read_tuple, input_node);
-      if (!expression) return nullptr;
+
+    for (const auto& groupby_expression : aggregate_node->group_by_expressions) {
+      const auto jit_expression =
+          _try_translate_expression_to_jit_expression(*groupby_expression, *read_tuples, input_node);
+      if (!jit_expression) return nullptr;
       // Create a JitCompute operator for each computed groupby column ...
-      if (expression->expression_type() != ExpressionType::Column) {
-        jit_operator->add_jit_operator(std::make_shared<JitCompute>(expression));
+      if (jit_expression->expression_type() != JitExpressionType::Column) {
+        jit_operator->add_jit_operator(std::make_shared<JitCompute>(jit_expression));
       }
       // ... and add the column to the JitAggregate operator.
-      aggregate->add_groupby_column(groupby_column.get<0>(), expression->result());
+      aggregate->add_groupby_column(groupby_expression->as_column_name(), jit_expression->result());
     }
 
-    for (const auto& aggregate_column : boost::combine(aggregate_column_names, aggregate_columns)) {
-      const auto aggregate_expression = aggregate_column.get<1>();
-      DebugAssert(aggregate_expression->type() == ExpressionType::Function, "Expression is not a function.");
-      const auto expression = _try_translate_expression_to_jit_expression(
-          *aggregate_expression->aggregate_function_arguments()[0], *read_tuple, input_node);
-      if (!expression) return nullptr;
+    for (const auto& expression : aggregate_node->aggregate_expressions) {
+      const auto aggregate_expression = std::dynamic_pointer_cast<AggregateExpression>(expression);
+      DebugAssert(aggregate_expression, "Expression is not a function.");
+
+      const auto jit_expression =
+          _try_translate_expression_to_jit_expression(*aggregate_expression->arguments[0], *read_tuples, input_node);
```

Comment:
`aggregate_expression ` can be `count`. Problematic is `count(*)` as it has no arguments. Found via trying to execute TPCH-1 with JIT.

Reply:
done!

Reply:
done but not pushed?

Reply:
pushed, but the change is in `_node_is_jittable`